src/app.js

```diff
@@ -1,35 +1,47 @@
 import React from "react";
+import {Provider} from 'react-redux';
+import {startMovieRequest} from './actions/movie';
 import ReactDOM from "react-dom";
+import configureStore from './store/configureStore';
 import {ErrorBoundary} from './components/ErrorBoundary';
 import SearchPage from './pages/SearchPage';
 import DetailsPage from './pages/DetailsPage';
 
 import 'normalize.css/normalize.css';
 import './styles.scss';
 
+const store = configureStore();
+
 class App extends React.Component {
   constructor(props) {
     super(props);
 
     // For shocasing purposes
     this.state = {
-      page: 'details'
+      page: 'search'
     };
   }
 
+  handleMovieSelected = (movieId) => {
```

Comment:
I would movie this to a separated file to control the "routing" and why not moving this into redux as well? Then you can have a kind of route component like this:

```javascript
export const Route = ({ currentPage, path, Component }) => (
   {(currentPage === path) ? <Component /> : null }
);

const mapStateToProps = (state) => ({ currentPage: state.router.currentPage });

export default connect(mapStateToProps)(Route);
```

Reply:
I didn't bother to do this because I knew the next lesson would be routing, so I didn't like the idea of implementing the routing in the state and then having to remove it. I just went for the easiest option.